database/migrations/2020_11_23_165933_create_skill_categories_table.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateSkillCategoriesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('skill_categories', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('key');
+            $table->json('name');
+            $table->integer('parent_category_id')->default(0)->nullable();
```

Comment:
Since its nullable, leave the default as null.

Reply:
@tristan-orourke Good catch, thanks. Looking at this again, I wonder if it's better to remove nullable and leave the default as 0 so that the model doesn't have a mixed type: thoughts?

Reply:
I actually prefer using null, its more unambiguous than having to know 0 represents no parent.

Reply:
@tristan-orourke The [documentation](https://backpackforlaravel.com/docs/4.0/crud-operation-reorder#requirements) for Backpack's Reorder feature (which is implemented in #4811) says to use a default value of 0. Did you want me to use a default value of null anyway?

Reply:
Okay, if we're using the reorder feature we should use their defaults.
I find it odd that they want to default to 0 but also make it nullable, so just leave a comment here and in the model referring to the Backpack feature.